docs/webhooks.rst

```diff
@@ -1,78 +1,128 @@
+automatically if you had previously set up a webhook with your provider
+manually.
 
-If your project is hosted on GitLab, you can easily add a hook that will rebuild
-your docs whenever you push updates.
+Authentication is required to use the generic endpoint. Either token-based
+authentication or basic authentication using your username and password can be
+used. Token-based authentication is recommended, you will find this token on the
+integration details page.
 
-* Go to the "Settings" page for your project
-* Click "Integrations"
-* In the "URL" section, enter ``https://readthedocs.org/gitlab``
-* Leave the default "Push events" selected
-* Click "Add Webhook"
+Parameters
+~~~~~~~~~~
 
-Others
-------
+This endpoint accepts the following arguments during an HTTP POST:
 
-Your ReadTheDocs project detail page has your post-commit hook on it; it will
-look something along the lines of ``http://readthedocs.org/build/<project_name>``.
-Regardless of which revision control system you use, you can just hit this URL
-to kick off a rebuild.
+branches
+    The names of the branches to trigger builds for. This can either be an array
+    of branch name strings, or just a single branch name string.
 
-The following parameters available to customize the behavior of custom webhooks:
+    Default: **latest**
 
-* ``'version_slug'``: The build version to trigger build for (defaults to ``'latest'``)
+token
+    The integration token. You'll find this value on the integration detail
+    page.
 
-  Example::
+For example, the cURL command to build the ``dev`` branch, using the token
+``1234``, would be::
 
-      $ curl -X POST --data "version_slug=$VERSION" https://readthedocs.org/build/$PROJECT_NAME
+    curl -X POST -d "branches=dev" -d "token=1234" https://readthedocs.org/api/v2/webhook/example-project/1/
 
-You could make this part of a hook using Git_, Subversion_, Mercurial_, or
-Bazaar_, perhaps through a simple script that accesses the build URL using
-``wget`` or ``curl``.
+A command like the one above could be called from a cron job or from a hook
+inside Git_, Subversion_, Mercurial_, or Bazaar_.
 
 .. _Git: http://www.kernel.org/pub/software/scm/git/docs/githooks.html
 .. _Subversion: http://mikewest.org/2006/06/subversion-post-commit-hooks-101
 .. _Mercurial: http://hgbook.red-bean.com/read/handling-repository-events-with-hooks.html
 .. _Bazaar: http://wiki.bazaar.canonical.com/BzrHooks
+
+Authentication
+~~~~~~~~~~~~~~
+
+This endpoint requires authentication. If authenticating with an integration
+token, a check will determine if the token is valid and matches the given
+project. If instead an authenticated user is used to make this request, a check
+will be performed to ensure the authenticated user is an owner of the project.
+
+Debugging webhooks
+------------------
+
+If you are experiencing problems with an existing webhook, you may be able to
+use the integration detail page to help debug the issue. Each project
```

Comment:
I'll see about adding some clarifying language. We can't link to anything useful here. This is were feature documentation, complete with images, is most useful.